### PR TITLE
WR #257591. Fixes #4. Added processing of all name-like values pulled from DB through urlencode().

### LINKED ISSUE #4: Special characters in activity names
Special chars (like %) aren't stripped out of name in the URL. If an activity ends in a % it will throw a fatal error.

Since we aren't actually using the name, we should just strip all but the most basic characters.

### PR DESCRIPTION
**Pull request by suankan**

### CODE REVIEW

File: tests/local_clean_urls.php [hunk 1]
```diff
@@ -202,6 +212,7 @@ public function test_local_cleanurls_simple() {
         $url = 'http://www.example.com/moodle/user/view.php?id=' . $this->staff->id . '&course=' . $this->course->id;
         $murl = new moodle_url($url);
         $clean = $murl->out();
+        var_dump($clean);
         $this->assertEquals('http://www.example.com/moodle/course/'.$this->course->shortname.'/user/' . $this->staff->username,
```
Comment (brendanheywood) [inline tests/local_clean_urls.php]: $this->course->shortname should be wrapped in urlencode($this->course->shortname)

File: tests/local_clean_urls.php [hunk 2]
```diff
@@ -153,19 +163,19 @@ public function test_local_cleanurls_simple() {
...
 
-        $url = 'http://www.example.com/moodle/course/view.php?id=' . $this->mancourse->id;
-        $murl = new moodle_url($url);
-        $clean = $murl->out();
-        $this->assertEquals('http://www.example.com/moodle/course/view?id=' . $this->mancourse->id, $clean,
-            "Clean: course is ignored because it's shortname clashes with dir or file");
-
-        $unclean = local_cleanurls\clean_moodle_url::unclean($clean)->raw_out();
-        $this->assertEquals($url, $unclean, "Unclean: course is ignored as clashed with php file");
+//        $url = 'http://www.example.com/moodle/course/view.php?id=' . $this->mancourse->id;
```
Comment (brendanheywood) [inline tests/local_clean_urls.php]: this looks like left over temp stuff??

Comment (suankan) [inline tests/local_clean_urls.php]: Yes I commented out this test case, I'll need to analyse it again.

File: tests/local_clean_urls.php (hunk 1 above)
Comment (suankan) [inline tests/local_clean_urls.php]: this was done.

**Pull request merged**